[WIP] [Tests] Optimized `pdo_mysql` tests

SakiTakamachi opened the pull request:
take2 of #12555

SakiTakamachi commented:
I don't have the concentration left, so I'll continue tomorrow.

SakiTakamachi commented:
done

Girgias commented on ext/pdo_mysql/tests/bug63185.phpt:
```diff
@@ -1,29 +1,30 @@
-$pdo->exec('CREATE PROCEDURE test_procedure_error_at_second ()
+$procedure = 'test_procedure_error_at_second_63185';
+
+$pdo->exec("DROP PROCEDURE IF EXISTS {$procedure}");
```
```suggestion
```
This line shouldn't be needed if the CLEAN section does its job

Girgias commented on ext/pdo_mysql/tests/bug76815.phpt:
```diff
@@ -4,38 +4,40 @@ Bug #76815: PDOStatement cannot be GCed/closeCursor-ed when a PROCEDURE resultse
+$procedure = 'bug76815_pdo_mysql_p';
 
-$st = $pdo->prepare('CALL tst2()');
+$pdo->query("DROP FUNCTION IF EXISTS {$func}");
+$pdo->query("DROP PROCEDURE IF EXISTS {$procedure}");
```
```suggestion
```
Should not be needed if CLEAN section does its job

Girgias commented on ext/pdo_mysql/tests/bug78152.phpt:
```diff
@@ -4,31 +4,32 @@ Bug #78152: PDO::exec() - Bad error handling with multiple commands
+
+var_dump($db->exec("INSERT INTO {$table} (id, label) VALUES (41, 'x'); INSERT INTO {$table}_bad (id, label) VALUES (42, 'y')"));
 $db->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
 try {
-    var_dump($db->exec("INSERT INTO test(id, label) VALUES (42, 'x'); INSERT INTO test_bad(id, label) VALUES (43, 'y')"));
+    var_dump($db->exec("INSERT INTO {$table} (id, label) VALUES (42, 'x'); INSERT INTO {$table}_bad (id, label) VALUES (43, 'y')"));
 } catch (PDOException $e) {
     echo $e->getMessage(), "\n";
 }
-
```
The warning version feels redundant? But that's by the by

SakiTakamachi replied:
In case of ERRMODE, it is not possible to check the return value of `exec()

Girgias commented on ext/pdo_mysql/tests/bug_41125.phpt:
```diff
@@ -4,29 +4,25 @@ Bug #41125 (PDO mysql + quote() + prepare() can result in segfault)
-if ($version < 40100)
-    die(sprintf("skip Need MySQL Server 5.0.0+, found %d.%02d.%02d (%d)\n",
-        $matches[1], $matches[2], $matches[3], $version));
+MySQLPDOTest::skipVersionThanLess(40100);
```
```suggestion
```
I think expecting a not totally outdated MySQL version is okay now

Girgias commented on ext/pdo_mysql/tests/bug_41997.phpt:
```diff
@@ -4,30 +4,22 @@ PDO MySQL Bug #41997 (stored procedure call returning single rowset blocks futur
-if ($version < 50000)
-    die(sprintf("skip Need MySQL Server 5.0.0+, found %d.%02d.%02d (%d)\n",
-        $matches[1], $matches[2], $matches[3], $version));
+MySQLPDOTest::skipVersionThanLess(50000);
```
ditto

Girgias commented on ext/pdo_mysql/tests/bug_41997.phpt:
```diff
@@ -4,30 +4,22 @@ PDO MySQL Bug #41997 (stored procedure call returning single rowset blocks futur
-$db->exec('CREATE PROCEDURE p() BEGIN SELECT 1 AS "one"; END');
+$procedure = 'bug_41997_pdo_mysql_p';
+
+$db->exec("DROP PROCEDURE IF EXISTS {$procedure}");
```
```suggestion
```
No need with CLEAN section

Girgias commented on ext/pdo_mysql/tests/bug_42499.phpt:
```diff
@@ -1,33 +1,19 @@
-if ($version < 41000)
-    die(sprintf("skip Need MySQL Server 4.1.0+, found %d.%02d.%02d (%d)\n",
-        $matches[1], $matches[2], $matches[3], $version));
+MySQLPDOTest::skipVersionThanLess(40100);
```
Ditto

Girgias commented on ext/pdo_mysql/tests/bug_44707.phpt:
```diff
@@ -1,34 +1,20 @@
-if ($version < 41000)
-    die(sprintf("skip Will work different with MySQL Server < 4.1.0, found %d.%02d.%02d (%d)\n",
-        $matches[1], $matches[2], $matches[3], $version));
+MySQLPDOTest::skipVersionThanLess(40100);
```
Ditto

Girgias commented on ext/pdo_mysql/tests/bug_50323.phpt:
```diff
@@ -42,19 +40,19 @@ $db = PDOTest::test_factory(__DIR__ . '/common.phpt');
+require_once __DIR__ . '/inc/mysql_pdo_test.inc';
+$db = MySQLPDOTest::factory();
 
 @$db->exec('DROP DATABASE IF EXISTS `crazy;dbname`');
```
```suggestion
$db->exec('DROP DATABASE IF EXISTS `crazy;dbname`');
```
Why is this error suppressed in the first place :(

Girgias commented on ext/pdo_mysql/tests/bug_61411.phpt:
```diff
@@ -1,29 +1,18 @@
-if ($version < 40106)
-    die(sprintf("skip Need MySQL Server 4.1.6+, found %d.%02d.%02d (%d)\n",
-        $matches[1], $matches[2], $matches[3], $version));
+MySQLPDOTest::skipVersionThanLess(40106);
```
Ditto

Girgias commented on ext/pdo_mysql/tests/bug_pecl_7976.phpt:
```diff
@@ -1,42 +1,30 @@
-if ($version < 50000)
-    die(sprintf("skip Need MySQL Server 5.0.0+, found %d.%02d.%02d (%d)\n",
-        $matches[1], $matches[2], $matches[3], $version));
+MySQLPDOTest::skipVersionThanLess(50000);
```
Ditto

Girgias commented on ext/pdo_mysql/tests/pdo_mysql_pconnect.phpt:
```diff
@@ -63,25 +62,25 @@ MySQLPDOTest::skip();
         $con1 = $tmp['_con1'];
 
-        @$db2 = new PDO($dsn, $user, $pass, array(PDO::ATTR_PERSISTENT => true));
+        @$db2 = new PDO($dsn, $user, $pass, [PDO::ATTR_PERSISTENT => true]);
```
```suggestion
        $db2 = new PDO($dsn, $user, $pass, [PDO::ATTR_PERSISTENT => true]);
```

Girgias reviewed (commented):
Did a partial review

SakiTakamachi commented:
Thank you, Fixed everything except for error suppression. There was an oversight, so I fixed that as well.

When it comes to error suppression, I first need to understand what it want to do with the test, and I look at it one by one. This will take some time, so please be patient.

SakiTakamachi commented on ext/pdo_mysql/tests/bug_44707.phpt:
```diff
@@ -1,34 +1,19 @@
-$db = PDOTest::test_factory(__DIR__ . '/common.phpt');
+require_once __DIR__ . '/inc/mysql_pdo_test.inc';
+$db = MySQLPDOTest::factory();
+$db->setAttribute(PDO::ATTR_STRINGIFY_FETCHES, true);
```
This attribute was set in PDOTest::test_factory, so the test will fail if it is not set.

Alternatively, there is a way to modify the expected value of the test.
